addresses_type.go

```diff
@@ -6,53 +6,55 @@ import (
+	Tags                      []ResourceTag `json:"tags,omitempty" doc:"the list of resource tags associated with ip address"`
+	VirtualMachineDisplayName string        `json:"virtualmachinedisplayname,omitempty" doc:"virtual machine display name the ip address is assigned to (not null only for static nat Ip)"`
+	VirtualMachineID          string        `json:"virtualmachineid,omitempty" doc:"virtual machine id the ip address is assigned to (not null only for static nat Ip)"`
+	Virtualmachinename        string        `json:"virtualmachinename,omitempty" doc:"virtual machine name the ip address is assigned to (not null only for static nat Ip)"`
+	VlanID                    string        `json:"vlanid,omitempty" doc:"the ID of the VLAN associated with the IP address. This parameter is visible to ROOT admins only"`
+	VlanName                  string        `json:"vlanname,omitempty" doc:"the VLAN associated with the IP address"`
+	VMIPAddress               net.IP        `json:"vmipaddress,omitempty" doc:"virutal machine (dnat) ip address (not null only for static nat Ip)"`
+	VpcID                     string        `json:"vpcid,omitempty" doc:"VPC the ip belongs to"`
+	ZoneID                    string        `json:"zoneid,omitempty" doc:"the ID of the zone the public IP address belongs to"`
+	ZoneName                  string        `json:"zonename,omitempty" doc:"the name of the zone the public IP address belongs to"`
 }
 
 // AssociateIPAddress (Async) represents the IP creation
 //
 // CloudStack API: https://cloudstack.apache.org/api/apidocs-4.10/apis/associateIpAddress.html
 type AssociateIPAddress struct {
-	Account    string `json:"account,omitempty"`
-	DomainID   string `json:"domainid,omitempty"`
-	ForDisplay *bool  `json:"fordisplay,omitempty"`
-	IsPortable *bool  `json:"isportable,omitempty"`
-	NetworkdID string `json:"networkid,omitempty"`
-	ProjectID  string `json:"projectid,omitempty"`
-	RegionID   string `json:"regionid,omitempty"`
-	VpcID      string `json:"vpcid,omitempty"`
-	ZoneID     string `json:"zoneid,omitempty"`
+	Account    string `json:"account,omitempty" doc:"the account to associate with this IP address"`
+	DomainID   string `json:"domainid,omitempty" doc:"the ID of the domain to associate with this IP address"`
+	ForDisplay *bool  `json:"fordisplay,omitempty" doc:"an optional field, whether to the display the ip to the end user or not"`
+	IsPortable *bool  `json:"isportable,omitempty" doc:"should be set to true if public IP is required to be transferable across zones, if not specified defaults to false"`
+	NetworkdID string `json:"networkid,omitempty" doc:"The network this ip address should be associated to."`
+	ProjectID  string `json:"projectid,omitempty" doc:"Deploy vm for the project"`
+	RegionID   int    `json:"regionid,omitempty" doc:"region ID from where portable ip is to be associated."`
```

Comment:
To be removed

Reply:
I guess regionId was the `int` representation name for `zoneid` which is a uuid.